code/game/atoms.dm

```diff
@@ -1392,6 +1394,24 @@
 
 		SEND_SIGNAL(src, COMSIG_ATOM_VV_MODIFY_TRANSFORM)
 
+	if(href_list[VV_HK_SPIN_ANIMATION] && check_rights(R_VAREDIT))
+		var/num_spins = input(usr, "How many spins?", "Spin Animation") as null|num
+		if(!num_spins)
+			return
+		var/spin_speed = input(usr, "How fast?", "Spin Animation") as null|num
+		if(!spin_speed)
+			return
+		var/direction = input(usr, "Which direction? (-1 = counter-clockwise, 1 = clockwise)", "Spin Animation") as null|num
+		if(!direction)
+			return
+		SpinAnimation(spin_speed, num_spins, direction)
+
+	if(href_list[VV_HK_STOP_ALL_ANIMATIONS] && check_rights(R_VAREDIT))
+		var/result = input(usr, "Are you sure?", "Stop Animating") in list("Yes", "No")
+		if(result == "No")
+			return
+		animate(src, transform = null, flags = ANIMATION_END_NOW) // Literally just fucking stop animating entirely. This tends to make stuff look bad, but admin said do the thing
```

Comment:
this doesn't account for potentially having nulls from `input()`, so in the name of proper UX, do it like this:
```suggestion
		if(result == "Yes")
			animate(src, transform = null, flags = ANIMATION_END_NOW) // Literally just fucking stop animating entirely. This tends to make stuff look bad, but admin said do the thing
		return
```

no reason to buy into the early return meme if an admin closes out of the prompt (because they weren't sure) and it does the action anyways.

Reply:
Good point.